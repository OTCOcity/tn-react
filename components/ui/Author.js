const Author = ({author}) => (
  DOM.div(
    {
      className: 'blog-list__item-author'
    },
    author

  )
);

Author.defaultProps = {
  author: 'Anonym'
};
Author.propTypes = {
  author: PropTypes.string
};
