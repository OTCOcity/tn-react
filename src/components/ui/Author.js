import React, {DOM, PropTypes} from 'react';
import Link from 'components/elements/Link';

import {postPath} from 'helpers/routes';

const Author = ({author, id}) => (
  DOM.div(
    {
      className: 'blog-list__item-author'
    },
    React.createElement(Link, {
      to: postPath(id),
      className: 'blog-list__item-author-link'
    }, author)
  )
);

Author.defaultProps = {
  author: 'Anonym'
};
Author.propTypes = {
  author: PropTypes.string
};

export default Author;
