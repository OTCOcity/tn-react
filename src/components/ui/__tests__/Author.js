import React from 'react';
import ReactDom from 'react-dom';

import {shallow} from 'enzyme';

import Link from 'components/elements/Link';

import Author from 'components/ui/Author';

describe('Author', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<Author/>, div);
  });

  describe('render', () => {
    it('should render Author name', () => {
      const itemProps = {id: 1, author: 'Ignat'};

      const item = shallow(<Author id={itemProps.id} author={itemProps.author} />);
      const header = <Link to="/posts/1" className='blog-list__item-author-link'>Ignat</Link>;


      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual Author', () => {
      const itemProps = {id: 1, author: 'Ignat'};
      const item = shallow(<Author id={itemProps.id} author={itemProps.author} />);

      expect(item).toMatchSnapshot();
    });
  });
});