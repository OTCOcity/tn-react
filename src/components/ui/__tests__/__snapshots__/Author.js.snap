// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author render should render usual Author 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="blog-list__item-author"
>
    <Link
        className="blog-list__item-author-link"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/posts/1"
    >
        Ignat
    </Link>
</div>,
  "nodes": Array [
    <div
      className="blog-list__item-author"
>
      <Link
            className="blog-list__item-author-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/posts/1"
      >
            Ignat
      </Link>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Author
        author="Ignat"
        id={1}
/>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "author": "Ignat",
          "id": 1,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="blog-list__item-author"
>
          <Link
                    className="blog-list__item-author-link"
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/posts/1"
          >
                    Ignat
          </Link>
</div>,
        "_debugID": 8,
        "_renderedOutput": <div
          className="blog-list__item-author"
>
          <Link
                    className="blog-list__item-author-link"
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/posts/1"
          >
                    Ignat
          </Link>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Author
    author="Ignat"
    id={1}
/>,
}
`;
